Feature/#10 youtube modal layout

hye410 opened the pull request:
## 💡 관련이슈
> #10 

## 🍀 작업 요약
1. Header 고정
2. 리스트에서 특정 상점 클릭 시 해당 상점에 대한 디테일 모달 레이아웃 작업
3. alert utils 생성

## 💬 리뷰 요구 사항
> [React-icons 라이브러리](https://react-icons.github.io/react-icons/)와 [sweetAlert2](https://sweetalert2.github.io/)를 설치했습니다.
처음에 pnpm install 한 번 받아주세요!
로컬에서 직접 모달을 보고싶다면 ? -> DetailModal  import 하셔서 확인해 보시면 됩니다!

![스크린샷 2025-02-27 오후 11 12 51](https://github.com/user-attachments/assets/7d239b7e-9119-4986-af89-72b9cf091c14)
추후에 제우님 메인 화면 작업 끝나시면 메인 화면에서 해당 모달 불러올 수 있도록
위에 첨부드린 사진처럼 코드 추가할 예정입니다~


## 특이사항

1.UI 구현은 해야하는데 데이터가 없어서, 우선 MOCK_DATA 만들어서 넣어놨습니다.
일단 일차적으로 UI에 대한 코드 리뷰 먼저 요청드립니다.

리뷰가 완료되면 dev에 merge하지 않고 다른 브랜치 따서 그 브랜치에 merge하겠습니다.
 ( dev에 merge 시 추후 api연결 후 충돌 예상 -> 새로 딴 다른 브랜치에서  api연결 작업 진행  )

2. sweetAlert는 utils에 openAlert라는 함수로 공통으로 뺐습니다.
제가 생각했을 땐, 성공/에러/안내/경고 정도면 충분할 거 같아서 우선 4가지의 타입만 작업해놨는데
더 필요한 타입 있으시면 추가해서 사용해 주시면 됩니다.
사용 예시는 다음과 같습니다.
`import { ALERT_TYPE } from '@/constants/alert-constant';`
`const { SUCCESS } =ALERT_TYPE; `
 `openAlert({ type: SUCCESS, text:"닉네임이 변경되었습니다.' });`

type은 상수로 정의해두었으니 거기서 꺼내쓰시면 됩니다. ( src/constants/alert-constant)
text는 alert에 보여질 문구를 의미합니다.
buttonText는 default로 '확인'으로 넣어놨습니다.
따라서 생략해도 무방하나 다른 문구를 넣고 싶다면 buttonText도 추가해 주세요!

## 💛 미리보기
> 
<img width="1582" alt="image" src="https://github.com/user-attachments/assets/35acf33c-5bde-4986-a1b8-1c900c5b9426" />
<img width="1452" alt="image" src="https://github.com/user-attachments/assets/65bc8b99-8330-4464-910d-b9ab522b9767" />



------------------------

의도치 않게 pr이 길어졌는데, 최최최종본입니다.^_^... 
디자인에 대한 비난은 모두 수용하겠습니다!

hye410 committed:
chore:#1 main.jsx 절대경로 추가
- merge하는 과정에서 App.jsx의 절대경로가 누락되어 추가함

hye410 committed:
[chore] Header 고정

hye410 committed:
feat:#10 디테일 모달 레이아웃 작업
- react-icons 설치 (북마크, 창닫기 부분에서 아이콘 필요)
- 아직 데이터가 없어서 MOCK_DATA로 주입

hye410 committed:
chore:#10 디테일 모달 색상 변경 및 상수 분리
- 모달 색상이 너무 구려서 팀원들 의견을 반영하여 바꿔보았습니다..
- supabase에서 받아오는 store key는 다른 영역에서도 사용하여 상수 파일로 분리
- 추후 다른 곳에서 modal을 사용할 경우 같은 css를 적용하고자 modal에 적용한 style을 global.css로 분리

hye410 committed:
chore:#10 모달 배경색 좀 더 밝게 변경
- 뒤에 지도 이미지 첨부해서 테스트 했더니 배경색이 좀 더 밝은게 더 나아보임

choichangyeon commented on src/components/modal/detail-modal.jsx:
```diff
@@ -0,0 +1,87 @@
+  //TODO: user정보 받아와서 bookmark 여부 확인 후 초기값으로 넣어주기
+  const [isBookMarked, setIsBookMarked] = useState(false);
+
+  const addToBookmark = () => {
+    setIsBookMarked(!isBookMarked);
+    alert(`북마크${isBookMarked ? '에서 삭제' : '에 추가'}되었습니다.`);
+    //TODO: alert 추후에 sweetAlert를 쓰든 바꾸기
+    //TODO: user의 bookmark 상태 바꾸기 (->post api연결)
+  };
```
저도 Alert에 대해서 라이브러리를 정하면 좋을 것 같습니다!

choichangyeon reviewed (approved):
수고하셨습니다!

JinchaeKim commented:
리액트 아이콘 라이브러리, 상수화 하는 부분, key 값을 주기 위해 div가 아닌 fragment 태그를 주는 부분 등 유용한 정보 많이 알아갑니다!!감사합니닷

hye410 committed:
feat:#10 sweetAlert 추가
- ALERT_TYPE constant로 정의
- openAlert로 util 분리
- detail-modal 기존의 window alert -> sweetAlert 적용

PomegranateBlue reviewed (approved):
확인했습니다

choichangyeon commented on src/lib/utils/openAlert.js:
```diff
@@ -0,0 +1,48 @@
+import { ALERT_TYPE } from '@/constants/alert-constant';
+import Swal from 'sweetalert2';
+
+const { ERROR, SUCCESS, INFO, WARNING } = ALERT_TYPE;
```
alert 타입을 나눠주신 부분 좋은것 같습니다! 잘 사용하도록 하겠습니다!

choichangyeon reviewed (commented):
수고하셨습니다!

mangomaneya commented on src/components/modal/detail-modal.jsx:
```diff
@@ -0,0 +1,88 @@
+import { FaRegBookmark } from 'react-icons/fa6';
+import { FaBookmark } from 'react-icons/fa';
+import { useState } from 'react';
+import { STORE_CONSTANT } from '@/constants/store-constant';
```
상수화로 더 일관된 ux를 유지하는데 큰 도움이 될것 같습니다 :)

mangomaneya reviewed (approved):
상수화를 통한 하드코딩을 최대한 막으려는 의도가 느껴집니다! 배울점이 많은 코드같습니다:)
